### ISSUE TITLE
When using ceedling options it appears the fff plugin no more works

### ISSUE THREAD

**Issue by ykoehler** [ElectronVector/fake_function_framework#15]
using "ceedling options:myopt1 test:all" result in 

```
Using fake function framework (fff)...

Test 'test_file.c'
----------------------
ceedling/test/test_file.c:1:17: error: fff.h: No such file or directory
ERROR: Shell command failed.
> Shell executed command:
'gcc -E -I"ceedling/test" -I"ceedling/test/support" -I"." -I".." -DTEST  -DTEST -DGNU_COMPILER "ceedling/test/test_file.c" -o "build/test/preprocess/files/test_file.c"'
> And exited with status: [1].

rake aborted!
ShellExecutionException: ShellExecutionException
.../tools/ceedling/lib/ceedling/tool_executor.rb:82:in `exec'
.../tools/ceedling/lib/ceedling/preprocessinator_file_handler.rb:12:in `preprocess_file'
.../tools/ceedling/lib/ceedling/preprocessinator.rb:40:in `preprocess_file'
.../tools/ceedling/lib/ceedling/preprocessinator.rb:12:in `block in setup'
.../x4/tools/ceedling/lib/ceedling/preprocessinator_helper.rb:33:in `preprocess_test_file'
.../tools/ceedling/lib/ceedling/preprocessinator.rb:25:in `preprocess_test_and_invoke_test_mocks'
.../tools/ceedling/lib/ceedling/test_invoker.rb:89:in `block in setup_and_invoke'
.../tools/ceedling/lib/ceedling/test_invoker.rb:63:in `each'
.../tools/ceedling/lib/ceedling/test_invoker.rb:63:in `setup_and_invoke'
.../tools/ceedling/lib/ceedling/rules_tests.rake:71:in `block (2 levels) in <top (required)>'
~/.rvm/gems/ruby-2.5.1/gems/ceedling-0.28.3/bin/ceedling:311:in `block in <top (required)>'
~/.rvm/gems/ruby-2.5.1/gems/ceedling-0.28.3/bin/ceedling:298:in `<top (required)>'
~/.rvm/gems/ruby-2.5.1/bin/ceedling:23:in `load'
~/.rvm/gems/ruby-2.5.1/bin/ceedling:23:in `<main>'
Tasks: TOP => test:test_file
(See full trace by running task with --trace)
ERROR: Ceedling Failed
```

Note that without the options:* it works nicely.

**Comment by sanko-oz** [comment]
I am facing the same problem. Somehow the fff plugin is not included when the two .yml files are merged and cmock is used to for mocking.

**Comment by sanko-oz** [comment]
Meanwhile I managed to figure out a dirty workaround. 

As far as I can see it, the problem is: 
the merge of the original `project.yml` and the `options.yml` happens after the first project setup in `rakefile.rb` . But after the merge another setup appears in `tasks_base.rake`, because `options:....` was set. 

The second setup is not re-instantiating any plugins, but it looks like its necessary for the fake_function_framework plugin, otherwise we fall back to cmock again. 

Here is what I did to make it work:
- locate `def load_plugin_scripts(script_plugins, system_objects)` in plugin_manager.rb
- surround `next if (@plugin_manager_helper.include?(@plugin_objects, plugin))` with an if:
```
if("#{plugin}.rb" != "fake_function_framework.rb")
  next if (@plugin_manager_helper.include?(@plugin_objects, plugin))
end
```
- then locate `def set_object` in diy.rb and comment-out following line:
``
raise "object '#{key}' already exists in context" if @cache.keys.include?(key)
``

These changes force the creation of a new fake_function_framework instance after the merge of the *.yml files. 

Another workaround without changing any ruby scripts is to remove fake_function_framework from `:plugins:` in your main `project.yml `and only add it to your `options.yml` .
But if you need to run certain test files without any options in parallel, its not that feasible.

Forgive me taking an apparent wrong approach, but I have no experience in ruby scripting and was only trying to keep my actual work going until a real fix is ready.  ; )

**Comment by mchernosky** [comment]
Fixed by #23
